src/core/net/ip6.cpp

```diff
@@ -1058,42 +1057,36 @@ otError Ip6::ProcessReceiveCallback(const Message &    aMessage,
         case kProtoUdp:
         {
             Udp::Header udp;
+            uint16_t    destPort;
+
             aMessage.Read(aMessage.GetOffset(), sizeof(udp), &udp);
 
-            switch (udp.GetDestinationPort())
-            {
-            case Mle::kUdpPort:
+            destPort = udp.GetDestinationPort();
 
+            if (destPort == Mle::kUdpPort)
+            {
                 // do not pass MLE messages
                 if (aMessageInfo.GetSockAddr().IsLinkLocal() || aMessageInfo.GetSockAddr().IsLinkLocalMulticast())
                 {
                     ExitNow(error = OT_ERROR_NO_ROUTE);
                 }
-
-                break;
-
-#if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE == 0
-            case kCoapUdpPort:
-
+            }
+#if !OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
+            if (destPort == kCoapUdpPort)
+            {
                 // do not pass TMF messages
                 if (Get<ThreadNetif>().IsTmfMessage(aMessageInfo))
                 {
                     ExitNow(error = OT_ERROR_NO_ROUTE);
                 }
-
-                break;
-#endif // OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
-
-            default:
-#if OPENTHREAD_FTD
-                if (udp.GetDestinationPort() == Get<MeshCoP::JoinerRouter>().GetJoinerUdpPort())
-                {
-                    ExitNow(error = OT_ERROR_NO_ROUTE);
-                }
+            }
 #endif
-                break;
+#if OPENTHREAD_FTD
```

Comment:
A quick not here:
I did not use `else if` in the code checking the `dstPort` values to cover the (unlikely by I think possible) situation that `GetJoinerUdpPort()` happen to match one of the ones above.